src/libraries/System.Runtime/tests/System/ByteTests.GenericMath.cs

```diff
@@ -13,6 +13,15 @@ public class ByteTests_GenericMath
         // IAdditionOperators
         //
 
+        private static byte NaNConversionOverflowValue()
+        {
+            if(System.Runtime.InteropServices.RuntimeInformation.ProcessArchitecture == System.Runtime.InteropServices.Architecture.RiscV64)
+            {
+                return (byte)0xFF;
+            }
```

Comment:
We're working on moving towards deterministic behavior across platforms: https://github.com/dotnet/runtime/issues/61885

This is going to necessitate, longer term, that saturation occurs as it is what most newer platforms do and what many specs (like WASM) require. This is also going to necessitate standardizing the behavior for `NaN`.

For x64, it currently returns a sentinel value `0x8000_0000` (int) or `0x8000_0000_0000_0000` (long) when the output type would overflow or when NaN is given. `byte` and `short` therefore currently expect `0` because its truncating the 32 or 64-bit result to 8 or 16-bits.

For Arm64, it saturates and converts `NaN` to zero. For WASM, it requires saturation and leaves `NaN` conversion undefined.

We plan on matching Arm64 for the cross platform deterministic behavior here as it is simple and straightforward. It matches the general IEEE 754 guidance that you compute the result as if to infinite precision and unbounded range, and then round to the nearest representable. `NaN` then logically has no equivalent, so it becoming zero "makes sense" and avoids confusion in other regards.

Reply:
As such, it would be better if we could ensure the RISC-V implementation adjusts and matches this behavior. The underlying platform specific behavior would then be available in the future for the APIs proposed in the linked issue.

Reply:
Thank you for your comment, I will try to find the proper solution.